src/main/java/com/solr/clientwrapper/domain/service/throttler/ThrottlerUtils.java

```diff
@@ -0,0 +1,13 @@
+package com.solr.clientwrapper.domain.service.throttler;
+
+public class ThrottlerUtils {
+	public static double formatRequestSizeStringToDouble(String size) {
+		String numericValue = size.substring(0, size.length()-2);
+		return Double.parseDouble(numericValue);
+	}
+	
+	public static double getSizeInkBs(String data) {
+		int numberOfChars = data.length();
+		return (8 * (int)((((numberOfChars) * 2) + 45) / 8))/1000;
```

Comment:
Why multiplied by 2?
Why first divide and then multiply?

Reply:
I had referred this resource- https://stackoverflow.com/questions/31206851/how-much-memory-does-a-string-use-in-java-8
But yeah, it should work by skipping mult. & div. by 8. I will check.

Reply:
Thank you for sharing the resource :)

Reply:
Ok, I checked it. So, there'd be a little difference if we remove those div & mult by 8. It ensures the **_rounded value for the memory allocation_** required for the given string data.
So, first divide, means get the _exact decimal calculation for a single byte_, then _parsed to (int) ensures that nearest whole number is considered_ before assigning memory, then multiply by 8, means all the bytes are considered.